Tasks/VsTest/distributedtest.ts

```diff
@@ -100,6 +82,17 @@ export class DistributedTest {
             utils.Helper.addToProcessEnvVars(envVars, 'DTA.CustomTestAdapters', this.dtaTestConfig.pathtoCustomTestAdapters);
         }
 
+        // Set proxy settings to environment if provided
+        if (!utils.Helper.isNullEmptyOrUndefined(this.dtaTestConfig.proxyConfiguration.proxyUrl)) {
+            utils.Helper.addToProcessEnvVars(envVars, 'DTA.ProxyUrl', this.dtaTestConfig.proxyConfiguration.proxyUrl);
+            utils.Helper.addToProcessEnvVars(envVars, 'DTA.ProxyUsername', this.dtaTestConfig.proxyConfiguration.proxyUserName);
+            utils.Helper.addToProcessEnvVars(envVars, 'DTA.ProxyPassword', this.dtaTestConfig.proxyConfiguration.proxyPassword);
+            utils.Helper.addToProcessEnvVars(envVars, 'DTA.ProxyBypassHosts', this.dtaTestConfig.proxyConfiguration.proxyBypassHosts);
+        }
+
+        // Adding Test Execution Host specific variables
+        await this.addDtaTestRunEnvVars(envVars);
```

Comment:
why this has to be async?

Reply:
because this method contains one call to updatesettings which is an async funtion due to historical reasons.